File: perl/2.pl
#!/usr/bin/perl
use List::Util qw(sum);
use strict;

# Find the sum of all the even-valued terms in the Fibonacci sequence 
# which do not exceed four million.

my @fibby = (1, 2);

while(@fibby[-1] < 4000000){
	push(@fibby, @fibby[-1] + @fibby[-2]);
}

print sum grep { ( $_ & 1 ) == 0 } @fibby;
print "\n";

File: haskell/2.hs
-- Find the sum of all the even-valued terms in the Fibonacci sequence.
-- which do not exceed four million.

main = do
    print $ sum $ filter even $ takeWhile (<4000000) fibs

fibs :: [Integer]
fibs = 1 : 1 : zipWith (+) fibs (tail fibs)
